src/project.rs

```diff
@@ -51,20 +67,55 @@ impl FileInfo {
         };
         Ok(file_info)
     }
+
+    pub fn blank() -> FileInfo {
+        FileInfo{ path: String::from("error"), lines: 0, extension: String::new() }
+    }
+
 }
 
+/// FileType marks what we determine a file to be.
+pub enum FileType {
+    Unknown,
+    Rust,
+    Go,
+    ELF,
+    Text,
+    Markdown,
+}
+
+impl Render for FileInfo {
+    fn render(&self) -> Markup {
+        html! {
+
+        }
+    }
+
+}
 pub fn analyze_go(path: &mut PathBuf, db: Arc<DB>) {
-    // ignore vendor
+    println!("analysis time");
+
+    // proj is our common prefix for all analysis
+    let proj = path.clone();
+    // batch_id is a DateTime, and will serve as our common prefix
+    let batch_id = Utc::now();
     // count number of .go files and the len of each
-    println!("wow!");
     let walker = WalkDir::new(path).into_iter();
     for entry in walker.filter_entry(|e| !is_hidden(e) && golang_files(e)) {
         // only walking Go files now...
-        let entry = entry.unwrap();
-        println!("name: {:?}, path: {:?}", entry.file_name(), entry.path());
+        println!("something happening");
+        let entry = entry.expect("could not get dir entry");
+        if entry.file_type().is_dir() {
+            println!("skipping analysis for dir");
+            continue;
+        }
+        println!("more stuff");
+        let k = make_key!(proj.to_str().expect("proj to string failed"), batch_id, entry.path().to_str().expect("path to string failed"));
 
-        // naive impl:
-        // "get" file repr from database
+        let fi = FileInfo::from_path(entry.path()).expect("could not make fileinfo");
+        println!("fileinfo made");
+        let encoded: Vec<u8> = serialize(&fi, Infinite).unwrap();
+        db.put(k.0.as_slice(), encoded.as_slice());
```

Comment:
What are the other values in the tuple of `k`?

Reply:
it's a struct, but the idea is the same: fixed width with static index to the field